website/blog/2021-08-05-win32component.md

```diff
@@ -0,0 +1,416 @@
+Now let's define the function that will invoke the native module to get the registry key and will store the result in the state:
+
+```javascript
+const getRegistryKey = async() => {
+  var result = await NativeModules.ReactNativeAppServiceModule.getRegistryKey(registryKeyName);
+  setRegistryKeyValue(result);
+}
+```
+
+The approach is the same as the previous one: the only difference is that, this time, we're invoking the `getRegistryKey()` function, which requires as parameter the name of the key we want to retrieve from the `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion` hive.
+
+Now let's build a minimal UI: a `TextInput` control, which will read the name of the registry key and store it in the state; a `Button`, which will invoke the `getRegistryKey()` function; a `Text`, which will display the value returned by the function.
+
+```javascript
+<View>
+  <TextInput onChangeText={text => setRegistryKeyName(text)} />
+  <Button title="Get registry key" onPress={getRegistryKey} />
+  <Text>{registryKeyValue}</Text>
+</View>
+```
+
+### Running the app
+
+We're ready to test our work! First, go to Visual Studio, right click on the Windows Application Packaging Project and choose Deploy. You might get the following error:
+
+```plaintext
+Error Task 'AddProjectMetadata' failed. The expression "[MSBuild]::MakeRelative(C:\ReactCSharp, *Undefined*)" cannot be evaluated. Illegal characters in path. C:\ReactCSharp\node_modules\react-native-windows\PropertySheets\Autolink.props
+ReactCSharp.Package C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\DesktopBridge\Microsoft.DesktopBridge.targets 408 
+Error MSB4184 The expression "[MSBuild]::MakeRelative(C:\ReactCSharp, *Undefined*)" cannot be evaluated. Illegal characters in path. C:\ReactCSharp\node_modules\react-native-windows\PropertySheets\Autolink.props ReactCSharp.Package C:\ReactCSharp\node_modules\react-native-windows\PropertySheets\Autolink.props 12
+```
+
+If that's a case, this is a known issue that has already been addressed, but the fix isn't available yet in the latest stable React Native for Windows version. However, the workaround is easy. Open the file in the path `node_modules\react-native-windows\PropertySheets\Autolink.props` and replace the following line:
+
+```xml
+<AutolinkCommandArgs Condition="'$(AutolinkCommandArgs)' == '' And '$(SolutionPath)' != '' And '$(ProjectPath)' != ''">--check --sln $([MSBuild]::MakeRelative($(AutolinkCommandWorkingDir), $(SolutionPath))) --proj $([MSBuild]::MakeRelative($(AutolinkCommandWorkingDir), $(ProjectPath)))</AutolinkCommandArgs>
+```
+
+with
+
+```xml
+<AutolinkCommandArgs Condition="'$(AutolinkCommandArgs)' == '' And '$(SolutionPath)' != '' And '$(SolutionPath)' != '*Undefined*' And '$(ProjectPath)' != ''">--check --sln $([MSBuild]::MakeRelative($(AutolinkCommandWorkingDir), $(SolutionPath))) --proj $([MSBuild]::MakeRelative($(AutolinkCommandWorkingDir), $(ProjectPath)))</AutolinkCommandArgs>
```

Comment:
same question for @dannyvv - this shouldn't be necessary after the backport, are we missing something else?

Reply:
Yeah, I remember it was backported, but I'm still hitting it. The sample I've published along the article is a new project I've created from scratch, using the latest version of RN and RNW.